toucan_connectors/google_sheets_2/google_sheets_2_connector.py

```diff
@@ -54,20 +57,41 @@ def get_form(cls, connector: 'GoogleSheets2Connector', current_config):
 Secrets = Dict[str, Any]
 
 
+class GoogleSheets2HiddenProperties(BaseModel):
+    """
+    This class contains dynamic hidden properties for Google Sheets 2 that are not flags
+
+    - secrets
+    """
+
+    secrets: Optional[Secrets]
+
+    def get_hidden_properties(self):
+        """Get hidden properties stored in this class."""
+        return {k: v for k, v in self.__dict__.items()}
+
+    def set_hidden_properties(self, hidden_properties):
+        """Set hidden properties to be stored in this class."""
+        for k, v in hidden_properties.items():
+            if hasattr(self, k):
+                setattr(self, k, v)
+
+
 class GoogleSheets2Connector(ToucanConnector):
     """The Google Sheets connector."""
 
+    # dynamic, non-hidden variables
     data_source_model: GoogleSheets2DataSource
-
-    auth_flow = 'oauth2'
-
     auth_flow_id: Optional[str]
 
-    # The following should be hidden properties
+    # flagging variable; required for initiating the oauth dance
+    _auth_flow = 'oauth2'
 
-    baseroute = 'https://sheets.googleapis.com/v4/spreadsheets/'
+    # static variable
+    _baseroute = 'https://sheets.googleapis.com/v4/spreadsheets/'
```

Comment:
Could we make that a real class property by the way ? That will save some db space and prevent to send it to the front-end